src/probability/martingale/optional_sampling.lean

```diff
@@ -0,0 +1,266 @@
+/-
+Copyright (c) 2023 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne
+-/
+
+import order.succ_pred.linear_locally_finite
+import probability.martingale.basic
+
+/-!
+# Optional sampling theorem
+
+If `τ` is a bounded stopping time and `σ` is another stopping time, then the value of a martingale
+`f` at the stopping time `min τ σ` is almost everywhere equal to
+`μ[stopped_value f τ | hσ.measurable_space]`.
+
+## Main results
+
+* `stopped_value_ae_eq_condexp_of_le_const`: the value of a martingale `f` at a stopping time `τ`
+  bounded by `n` is the conditional expectation of `f n` with respect to the σ-algebra generated by
+  `τ`.
+* `stopped_value_ae_eq_condexp_of_le`: if `τ` and `σ` are two stopping times with `σ ≤ τ` and `τ` is
+  bounded, then the value of a martingale `f` at `σ` is the conditional expectation of its value at
+  `τ` with respect to the σ-algebra genrated by `σ`.
+* `stopped_value_min_ae_eq_condexp` : the optional sampling theorem: if `τ` is a bounded stopping
+  time and `σ` is another stopping time, then the value of a martingale `f` at the stopping time
+  `min τ σ` is almost everywhere equal to `μ[stopped_value f τ | hσ.measurable_space]`.
```

Comment:
```suggestion
  `min τ σ` is the conditional expectation of its value at `τ` with respect to the σ-algebra generated by `σ`.
```
Since you use sentences to describe the other results, I would use a similar sentence here as well, to emphasize that the results are similar (and that each result is a consequence of the next one).